test/test_nn.py

```diff
@@ -16404,6 +16404,123 @@ def _helper(zero_infinity):
         _helper(zero_infinity=True)
         _helper(zero_infinity=False)
 
+    def _CTCLoss_no_batch_dim_helper(self, device, input_length, vocab_size, target_length):
+        batch_size = 1
+
+        log_probs = torch.randn(input_length, batch_size, vocab_size, dtype=torch.float, device=device) \
+                         .log_softmax(2).requires_grad_()
+        targets = torch.randint(low=1, high=vocab_size - 1, size=(batch_size, target_length), dtype=torch.int, device=device)
+        input_lengths = batch_size * [input_length]
+        target_lengths = batch_size * [target_length]
+
+        log_probs_no_bd = log_probs.squeeze(1).detach().clone().requires_grad_()
+        targets_no_bd = targets.squeeze(0).detach().clone()
+        input_lengths_no_bd = torch.tensor(input_length)
+        target_lengths_no_bd = torch.tensor(target_length)
+
+        return ((log_probs, targets, input_lengths, target_lengths),
+                (log_probs_no_bd, targets_no_bd, input_lengths_no_bd, target_lengths_no_bd))
+
+    def _CTCLoss_check_helper(self, log_probs_refs, log_probs_no_bd_refs, losses, losses_no_bd, reduction, input_length, vocab_size):
+        def _check_list(expected, list_to_compare, atol=None, rtol=None):
+            for ele in list_to_compare:
+                self.assertEqual(expected, ele, atol=atol, rtol=rtol)
+        
+        _check_list((1,) if reduction == 'none' else (), [loss.shape for loss in losses])
+        _check_list((), [loss.shape for loss in losses_no_bd])
+
+        _check_list((input_length, 1, vocab_size), [t.grad.shape for t in log_probs_refs])
+        _check_list((input_length, vocab_size), [t.grad.shape for t in log_probs_no_bd_refs])
+
+        _check_list(losses[0], losses[1:], atol=1e-4, rtol=0)
+        _check_list(losses[0].squeeze(0), losses_no_bd, atol=1e-4, rtol=0)
+
+        _check_list(log_probs_refs[0].grad, [t.grad for t in log_probs_refs[1:]], atol=1e-4, rtol=0)
+        _check_list(log_probs_refs[0].grad.squeeze(1), [t.grad for t in log_probs_no_bd_refs[1:]], atol=1e-4, rtol=0)
```

Comment:
@jbschlosser This function checks each combination of the size/values of the losses/gradients against each other